### DISCUSSION TITLE
What are google services needed for?

### DISCUSSION THREAD

**Discussion by schmadde** [NightscoutFoundation/xDrip#1994] [General]
I think this has been asked before some time in the gitter channel maybe, but I stumbled over it now and do not remember the outcome.

I have an android device on which I really want to have a data collection service and BG display for Dexcom G6 running so I tried to sideload xdrip+ to it. Unfortunately this is an "unofficial" Android device and does not have google services. I assume this is a problem for other people as well who want to run it on something not being a smartphone or not an official one. 

xdrip+ warns that it needs google services to run - and indeed it does not run without it. 

For what parts is google services needed and would it be feasable to reimplement stuff that is needed just for the use case "collect Data from Dexcom G6 and display a value" ?

**Comment by jonprogers** [comment]
I think I remember reading somewhere that xDrip followers transfer data via google services - I don't know enough to have a view about anything where it might be possible or desirable to reimplement functionality in a different way, sorry.

**Reply by schmadde** [reply]
I have no need for follower support and also not for maybe 90% of what xdrip+ can do. Since this device (a bicycle computer) has no network connectivity I also do not need any Nightscout integration. 

I just wanted to know if something in sensor readout and data display (and also notification or statusbar display) needs google services and how it can be worked around. 

I saw that the device could communicate with the sensor somehow, but could not read any values

**Reply by jonprogers** [reply]
You need input from someone who knows way more than I do then! Sorry.

**Comment by tolot27** [comment]
Did you find and read #1379?
